examples/usb/examples/serial.rs

```diff
@@ -0,0 +1,79 @@
+#![no_std]
+#![no_main]
+
+use panic_semihosting as _;
+
+use cortex_m_rt::entry;
+use nrf52840_hal::usbd::Usbd;
+use nrf52840_hal::clocks::Clocks;
+use nrf52840_pac::Peripherals;
+use usb_device::device::{UsbDeviceBuilder, UsbVidPid};
+use usbd_serial::{SerialPort, USB_CLASS_CDC};
+
+
+#[entry]
+fn main() -> ! {
+    let periph = Peripherals::take().unwrap();
+    while !periph
+        .POWER
+        .usbregstatus
+        .read()
+        .vbusdetect()
+        .is_vbus_present()
+    {}
+
+    // wait until USB 3.3V supply is stable
+    while !periph
+        .POWER
+        .events_usbpwrrdy
+        .read()
+        .events_usbpwrrdy()
+        .bit_is_clear()
```

Comment:
This waits for the bit to be 0. Shouldn't it wait for 1 instead?

Reply:
Changing it to wait for 1 makes it wait forever :rofl:. It seems it's not necessary anyway.

Reply:
I also don't think this code is not needed as devices should work without VBUS detection mechanism.

Reply:
I have a board where 5v is permanently wired to VBUS (so the nrf thinks usb is always plugged in) and yes, it still works fine, including hotplugging.

Reply:
Removed useless code, thank you :)